### PR TITLE
Refactor to improve readability

### LINKED ISSUE #418: Calling extension.ts -> activate() method fails in 0.15.0
There is a regression in 0.15.0 likely due to these changes - https://github.com/redhat-developer/vscode-xml/commit/5495a0e2e725a61d58c49af14b94e5d0a1312e6e.

The following code used to work in 0.14.0, but now extensionApi is 'undefined'.

```
const redHatExtension = vscode.extensions.getExtension('redhat.vscode-xml');
const extensionApi = await redHatExtension!.activate();
```

### PR DESCRIPTION
**Pull request by datho7561**
Use `await` in `extension.ts`. Move some functionality out of `extension.ts`. Move some TypeScript files into subfolders. Fix `.editorconfig`.

Fixes #418

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -140,304 +57,130 @@ namespace ActionableNotification {
...
-
-            setupActionableNotificationListener(languageClient);
+  const serverOptions: Executable = await prepareExecutable(
+    requirementsData, collectXmlJavaExtensions(extensions.all, getXMLConfiguration().get("extension.jars", [])), context);
 
-            // Handler for 'xml/executeClientCommand` request message that executes a command on the client
-            languageClient.onRequest(ExecuteClientCommandRequest.type, async (params: ExecuteCommandParams) => {
-              return await commands.executeCommand(params.command, ...params.arguments);
-            });
+  languageClient = new LanguageClient('xml', 'XML Support', serverOptions, clientOptions);
```
Comment (angelozerr) [inline src/extension.ts:118]: I will move all LSP client code into the `client/xmlClient.ts` like HTML https://github.com/microsoft/vscode/blob/master/extensions/html-language-features/client/src/htmlClient.ts and I will move tagClosing.ts in this `client` folder.

**Comment by angelozerr** [comment]
@datho7561 do you think your PR will fix https://github.com/redhat-developer/vscode-xml/issues/418 ?

File: src/extension.ts (hunk 1 above)
Comment (angelozerr) [reply src/extension.ts:118]: IMHII think it should be nice to have the following folders:

 * `client` 
 * `server`

to be consistent with HTML language server for instance https://github.com/microsoft/vscode/tree/main/extensions/html-language-features

**Comment by datho7561** [comment]
> @datho7561 do you think your PR will fix #418 ?

It might; I haven't had time to test it yet.

File: src/commands/registerCommands.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+/**
+ * Register the commands for vscode-xml
+ *
+ * @param context the extension context
+ */
+export async function registerCommands(context: ExtensionContext, languageClient: LanguageClient) {
+
+  // XML Documentation commands
+  context.subscriptions.push(markdownPreviewProvider);
```
Comment (angelozerr) [inline src/commands/registerCommands.ts:15]: Create a new function registerDocsCommands and move the docs code to this function

File: src/commands/registerCommands.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  context.subscriptions.push(commands.registerCommand(CommandConstants.OPEN_DOCS, async (params: { page: string, section: string }) => {
+    const page = params.page.endsWith('.md') ? params.page.substr(0, params.page.length - 3) : params.page;
+    const uri = page + '.md';
+    const sectionId = params.section || '';
+    const title = 'XML ' + page;
+    markdownPreviewProvider.show(context.asAbsolutePath(path.join('docs', uri)), title, sectionId, context);
+  }));
+
+  // Code Lens actions
+  context.subscriptions.push(commands.registerCommand(CommandConstants.SHOW_REFERENCES, (uriString: string, position: Position) => {
```
Comment (angelozerr) [inline src/commands/registerCommands.ts:31]: Create a new function registerCodeLensCommands and move the codeLens code to this function

File: src/commands/registerCommands.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      // Consume references service from the XML Language Server
+      let param = languageClient.code2ProtocolConverter.asTextDocumentPositionParams(document, position);
+      languageClient.sendRequest(ReferencesRequest.type, param).then(locations => {
+        commands.executeCommand(CommandConstants.EDITOR_SHOW_REFERENCES, uri, languageClient.protocol2CodeConverter.asPosition(position), locations.map(languageClient.protocol2CodeConverter.asLocation));
+      })
+    })
+  }));
+
+  // Revalidate current file
+  context.subscriptions.push(commands.registerCommand(CommandConstants.VALIDATE_CURRENT_FILE, async (params) => {
```
Comment (angelozerr) [inline src/commands/registerCommands.ts:43]: Create a new function registerValidateCommands and move the validate code to this function

**Review by angelozerr — APPROVED** [review]

**Comment by angelozerr** [comment]
It works like a charm on Windows OS, great job @datho7561 !

**Pull request merged**